File: anyharness/sdk/src/types/replay.ts
import type { components } from "../generated/openapi.js";
import type { Session } from "./sessions.js";

type GeneratedCreateReplaySessionResponse =
  components["schemas"]["CreateReplaySessionResponse"];

export type ReplayRecordingSummary =
  components["schemas"]["ReplayRecordingSummary"];
export type ListReplayRecordingsResponse =
  components["schemas"]["ListReplayRecordingsResponse"];
export type ExportReplayRecordingRequest =
  components["schemas"]["ExportReplayRecordingRequest"];
export type ExportReplayRecordingResponse =
  components["schemas"]["ExportReplayRecordingResponse"];
export type CreateReplaySessionRequest =
  components["schemas"]["CreateReplaySessionRequest"];
export type CreateReplaySessionResponse = Omit<
  GeneratedCreateReplaySessionResponse,
  "session"
> & {
  session: Session;
};
export type AdvanceReplaySessionResponse =
  components["schemas"]["AdvanceReplaySessionResponse"];

File: anyharness/crates/anyharness-contract/src/v1/replay.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

use super::Session;

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct ReplayRecordingSummary {
    pub id: String,
    pub label: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub source_session_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct ListReplayRecordingsResponse {
    pub recordings: Vec<ReplayRecordingSummary>,
}

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct ExportReplayRecordingRequest {
    pub session_id: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct ExportReplayRecordingResponse {
    pub recording: ReplayRecordingSummary,
}

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct CreateReplaySessionRequest {
    pub workspace_id: String,
    pub recording_id: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub speed: Option<f32>,
}

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct CreateReplaySessionResponse {
    pub session: Session,
}

#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct AdvanceReplaySessionResponse {
    pub advanced: bool,
}
